app/controllers/BankDetailsController.scala

```diff
@@ -99,31 +102,47 @@ class BankDetailsController @Inject() (
   }
 
   val enterBankDetailsSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
-    BankDetailsForm.form
-      .bindFromRequest()
-      .fold(
-        formWithErrors =>
-          Future.successful(Ok(views.enterBankDetailsPage(formWithErrors, BankDetailsController.chooseTypeOfAccountUrl))),
-        (bankDetailsForm: BankDetailsForm) => {
-
-          val directDebitDetails: DirectDebitDetails = DirectDebitDetails(
-            BankDetails(
-              name          = bankDetailsForm.name,
-              sortCode      = bankDetailsForm.sortCode,
-              accountNumber = bankDetailsForm.accountNumber
-            ),
-            bankDetailsForm.isSoleSignatory.asBoolean
-          )
-
-          journeyService.updateDirectDebitDetails(request.journeyId, directDebitDetails)
-            .map { _ =>
-              bankDetailsForm.isSoleSignatory match {
-                case IsSoleSignatoryFormValue.Yes => Redirect(routes.BankDetailsController.checkBankDetails)
-                case IsSoleSignatoryFormValue.No  => Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage)
-              }
+    val validForm = BankDetailsForm.form.bindFromRequest()
+    validForm.fold(
+      hasErrors = (formWithErrors: Form[BankDetailsForm]) =>
+        Future.successful(Ok(views.enterBankDetailsPage(formWithErrors, BankDetailsController.chooseTypeOfAccountUrl))),
+
+      success = (bankDetailsForm: BankDetailsForm) => {
+
+        val directDebitDetails: DirectDebitDetails = DirectDebitDetails(
+          BankDetails(
+            name          = bankDetailsForm.name,
+            sortCode      = bankDetailsForm.sortCode,
+            accountNumber = bankDetailsForm.accountNumber
+          ),
+          bankDetailsForm.isSoleSignatory.asBoolean
+        )
+
+          def enterBankDetailsPageWithBarsError(error: FormError): Result = {
+            Ok(views.enterBankDetailsPage(
+              validForm.withError(error),
+              BankDetailsController.chooseTypeOfAccountUrl
+            ))
+          }
+
+        journeyService.updateDirectDebitDetails(request.journeyId, directDebitDetails)
+          .flatMap { _ =>
```

Comment:
it feels like we should also be updating the BE with the BARs result so we can check on subsequent pages what the result is. We wouldn't want allow someone to type in the URL of the check bank details page on the browser and see that page sucessully if the BARs result is negative

Reply:
Yep. I think in the next ticket when we call the verify endpoint we will need to store state/stage in the BE. I was going to address that there. What do yuo think?